dll/win32/shlwapi/utils.cpp

```diff
@@ -115,3 +133,93 @@ PathFileExistsDefExtAndAttributesW(
 
     return TRUE;
 }
+
+static inline BOOL
+SHLWAPI_IsBogusHRESULT(HRESULT hr)
+{
+    return (hr == E_FAIL || hr == E_INVALIDARG || hr == E_NOTIMPL);
+}
+
+/*************************************************************************
+ * IShellFolder_GetDisplayNameOf [SHLWAPI.316]
+ */
+EXTERN_C HRESULT WINAPI
+IShellFolder_GetDisplayNameOf(
+    _In_ IShellFolder *psf,
+    _In_ LPCITEMIDLIST pidl,
+    _In_ DWORD uFlags,
+    _Out_ LPSTRRET lpName,
+    _In_ DWORD dwRetryFlags)
+{
+    HRESULT hr;
+
+    TRACE("(%p, %p, 0x%lX, %p, 0x%lX)\n", psf, pidl, uFlags, lpName, dwRetryFlags);
+
+    hr = psf->GetDisplayNameOf(pidl, uFlags, lpName);
+    if (!SHLWAPI_IsBogusHRESULT(hr))
+        return hr;
+
+    dwRetryFlags |= 0x80000000;
+
+    if ((uFlags & SHGDN_FORPARSING) == 0)
+        dwRetryFlags |= 1;
```

Comment:
So this flag can be named something like RETRYWITHFORPARSING?

Reply:
And I assume you need to try to call the same method again after adjusting the SHGDN flags?

Reply:
Yes, something like this.

Reply:
Fixed.